packages/lesswrong/lib/collections/comments/helpers.js

```diff
@@ -32,20 +13,23 @@ Comments.getAuthorName = function (comment) {
   return user ? Users.getDisplayName(user) : comment.author;
 };
 
-// LessWrong Helpers
-
 /**
  * @summary Get URL of a comment page.
  * @param {Object} comment
  */
 // LW: Overwrite the original example-forum Comments.getPageUrl
-Comments.getPageUrl = function(comment, isAbsolute = false){
+Comments.getPageUrl = function(comment, isAbsolute = false) {
   const post = Posts.findOne(comment.postId);
   return `${Posts.getPageUrl(post, isAbsolute)}#${comment._id}`;
 };
 
+Comments.getPageUrlFromIds = function(postId, postSlug, commentId, isAbsolute=false) {
```

Comment:
Looking at this more closely, I don't think this is a good idea. It would require the karma-changes query to have an extra join against the posts table, which is a pretty major significant performance cost, and the only upside would be to make the link URL slightly more aesthetic.

Reply:
The issue is that `Posts.getPageUrl` customizes the link using additional fields (isEvent, groupId, slug), which we aren't fetching, and which are expensive to fetch. (I'd estimate it's around a 2x increase in overall database disk I/O, for the site as a whole.) Given how high impact this feature is and that we've already had to roll it back once for speed issues, I really don't want to take on this risk.